tests/main/refresh-app-awareness-notify/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: Ensure that refresh app-awareness notifications work
+
+details: |
+    Test that notifications for continued auto-refreshes are delivered
+
+# Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+systems: [-ubuntu-14.04-*]
+
+restore: |
+    snap unset system refresh.timer
+
+execute: |
+    # setup test snap and make sure an app keeps it busy
+    snap install test-snapd-sh
+    test-snapd-sh.sh -c 'exec sleep infinity' &
+    PID="$!"
+    # Note that test-snapd-sh has different revisions in stable and edge
+    snap switch --edge test-snapd-sh
+
+    # trigger auto-refresh
+    snap unset system refresh.hold
+    systemctl stop snapd.{service,socket}
+    "$TESTSTOOLS"/snapd-state force-autorefresh
+    systemctl start snapd.{socket,service}
+    retry -n 12 --wait 10 sh -c 'journalctl -u snapd | MATCH "notifying agents about pending refresh for snap \"test-snapd-sh\""'
+
+    # exit test-snap, this should trigger the refresh right away
+    kill "$PID"
+
+    retry -n 12 --wait 10 sh -c 'journalctl -u snapd | MATCH "notifying user client about continued refresh for \"test-snapd-sh\""'
```

Comment:
nitpick: a 10 second wait to retry might be overkill

Reply:
Thanks, I changed to 60s in single second lookups